src/gql/schema.rs

```diff
@@ -0,0 +1,626 @@
+use std::collections::BTreeMap;
+
+use async_graphql::dynamic::TypeRef;
+use async_graphql::dynamic::{Enum, Type};
+use async_graphql::dynamic::{Field, Interface};
+use async_graphql::dynamic::{FieldFuture, InterfaceField};
+use async_graphql::dynamic::{InputObject, Object};
+use async_graphql::dynamic::{InputValue, Schema};
+use async_graphql::indexmap::IndexMap;
+use async_graphql::Name;
+use async_graphql::Value as GqlValue;
+use serde_json::Number;
+use surrealdb::err::Error;
+use surrealdb::sql;
+use surrealdb::sql::statements::SelectStatement;
+use surrealdb::sql::Expression;
+use surrealdb::sql::Kind;
+use surrealdb::sql::{Cond, Fields};
+use surrealdb::sql::{Statement, Thing};
+
+use super::ext::IntoExt;
+use crate::dbs::DB;
+use crate::gql::utils::GqlValueUtils;
+use surrealdb::kvs::LockType;
+use surrealdb::kvs::TransactionType;
+use surrealdb::sql::Value as SqlValue;
+
+macro_rules! limit_input {
+	() => {
+		InputValue::new("limit", TypeRef::named(TypeRef::INT))
+	};
+}
+
+macro_rules! start_input {
+	() => {
+		InputValue::new("start", TypeRef::named(TypeRef::INT))
+	};
+}
+
+macro_rules! id_input {
+	() => {
+		InputValue::new("id", TypeRef::named_nn(TypeRef::STRING))
+	};
+}
+
+macro_rules! order {
+	(asc, $field:expr) => {{
+		let mut tmp = ::surrealdb::sql::Order::default();
+		tmp.order = $field.into();
+		tmp.direction = true;
+		tmp
+	}};
+	(desc, $field:expr) => {{
+		let mut tmp = ::surrealdb::sql::Order::default();
+		tmp.order = $field.into();
+		tmp
+	}};
+}
+
+pub async fn get_schema() -> Result<Schema, Box<dyn std::error::Error>> {
+	let kvs = DB.get().unwrap();
+	let mut tx = kvs.transaction(TransactionType::Read, LockType::Optimistic).await?;
+	let tbs = tx.all_tb("test", "test").await?;
+	let mut query = Object::new("Query");
+	let mut types: Vec<Type> = Vec::new();
+	// TODO: remove hardcoded db and ns
```

Comment:
Do we want to merge it in with the hardcoded ns & db?

Reply:
Ideally not, I'm looking into it, but I think it's quite a bit of work to add, as it requires rewriting the web server integration